File: Models/RecurringExpense.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MilaAPI.Models
{
	public class RecurringExpense
	{
		[Key]
		public int Id { get; set; }

		[Required]
		public int UserId { get; set; }  // Related to a User
		public User User { get; set; }

		[Required]
		public int CategoryId { get; set; }  // Related to a Category
		public Category Category { get; set; }

		[Column(TypeName = "decimal(18,2)")]
		public decimal Amount { get; set; }
		public DateTime StartDate { get; set; }
		public DateTime? EndDate { get; set; }  // Optional if recurring expense ends at some point

		public RecurrenceType Recurrence { get; set; }  // Daily, Weekly, Monthly, Yearly

		public bool IsActive { get; set; }  // Determines if the recurring expense is active
		public DateTime NextDueDate { get; set; }
		public string Name { get; set; }
	}

	public enum RecurrenceType
	{
		Daily,
		Weekly,
		Monthly,
		Yearly
	}
}

File: front-end/milaapp/lib/models/recurring_expense.dart
class RecurringExpense {
  final String id;
  final double amount;
  final String description;
  final DateTime nextDueDate;
  final String userId;
  final String categoryId;

  RecurringExpense({
    required this.id,
    required this.amount,
    required this.description,
    required this.nextDueDate,
    required this.userId,
    required this.categoryId,
  });

  factory RecurringExpense.fromJson(Map<String, dynamic> json) {
    return RecurringExpense(
      id: json['id'],
      amount: json['amount'],
      description: json['description'],
      nextDueDate: DateTime.parse(json['nextDueDate']),
      userId: json['userId'],
      categoryId: json['categoryId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'amount': amount,
      'description': description,
      'nextDueDate': nextDueDate.toIso8601String(),
      'userId': userId,
      'categoryId': categoryId,
    };
  }
}
